### PR TITLE
#215/Create an endpoint to unblock a user

### LINKED ISSUE #212: Create a database method to delete a block

### LINKED ISSUE #215: Create an endpoint to unblock a user
Create a `DELETE /users/block/{userId}` endpoint to unblock a user

### PR DESCRIPTION
**Pull request by gbudau**
#212/Create a database method to delete a block
Add ParseUUIDPipe to validate the user ID to the block endpoints

close #212
close #215

### CODE REVIEW

**Review by vdedios — APPROVED** [review]
Neat!

File: transcendence-app/src/user/infrastructure/db/postgres/block.postgres.repository.ts [hunk 1]
```diff
@@ -31,4 +31,18 @@ export class BlockPostgresRepository
...
+
+  async deleteBlock(
+    blockerId: string,
+    blockedId: string,
+  ): Promise<BlockEntity | null> {
+    const block = await makeQuery<BlockEntity>(this.pool, {
+      text: `DELETE
+      FROM ${this.table}
+      WHERE ${BlockKeys.BLOCKER_ID} = $1 AND ${BlockKeys.BLOCKED_ID} = $2
+      RETURNING *;`,
```
Comment (Damrod) [inline transcendence-app/src/user/infrastructure/db/postgres/block.postgres.repository.ts:43]: what does the statement RETURNING in this context do?

**Review by Damrod — APPROVED** [review]
thank you!
LGTM

File: transcendence-app/src/user/infrastructure/db/postgres/block.postgres.repository.ts (hunk 1 above)
Comment (gbudau) [reply transcendence-app/src/user/infrastructure/db/postgres/block.postgres.repository.ts:43]: By using [`RETURNING *`](https://www.postgresql.org/docs/15/dml-returning.html), Postgres returns the content of the deleted rows.
We could also use the [`rowcount`](https://node-postgres.com/api/result#resultrowcount-int) variable from the `pg.Result`.

**Comment by gbudau** [comment]
Thanks!

**Pull request merged**